packages/angular_devkit/core/node/host_spec.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+// tslint:disable:no-any
+// tslint:disable:non-null-operator
+import { normalize, virtualFs } from '@angular-devkit/core';
+import { NodeJsAsyncHost } from '@angular-devkit/core/node';
+import * as fs from 'fs';
+import { Observable } from 'rxjs/Observable';
+import { Subscription } from 'rxjs/Subscription';
+
+const temp = require('temp');
+
+
+describe('NodeJsAsyncHost', () => {
+  let root: string;
+  let host: virtualFs.Host<fs.Stats>;
+
+  beforeEach(() => {
+    root = temp.mkdirSync('core-node-spec-');
+    host = new virtualFs.ScopedHost(new NodeJsAsyncHost(), normalize(root));
+  });
+  afterEach(done => {
+    host.delete(normalize('/'))
+      .subscribe({ complete() { done(); } });
+  });
+
+  it('can watch', done => {
+    let obs: Observable<virtualFs.HostWatchEvent>;
+    let subscription: Subscription;
+    const content = virtualFs.stringToFileBuffer('hello world');
+    const content2 = virtualFs.stringToFileBuffer('hello world 2');
+    const allEvents: virtualFs.HostWatchEvent[] = [];
+
+    Promise.resolve()
+      .then(() => fs.mkdirSync(root + '/sub1'))
+      .then(() => fs.writeFileSync(root + '/sub1/file1', 'hello world'))
+      .then(() => new Promise(resolve => setTimeout(resolve, 1000)))
```

Comment:
On Angular CLI I've seen watchers register writes several seconds after they happen. Would not be surprised to have this flake out randomly. Also am not sure what the solution is.

Reply:
There's no solution, this is at best a hack. The truth is I get like 5 events here, even with the `Sync` fs operations, on both Mac and Linux.

We can only verify that there is some event registered. Maybe I should watch first, then discard everything, then do some operations and check those. Let me experiment.